GCRCatalogs/register.py

```diff
@@ -3,24 +3,77 @@
 import warnings
 import yaml
 import requests
+import socket
 from GCR import BaseGenericCatalog
 
-
-__all__ = ['has_catalog', 'get_catalog_config', 'get_available_catalogs', 'load_catalog']
+__all__ = ['has_catalog', 'get_catalog_config', 'get_available_catalogs', 'load_catalog', 'get_root_dir', 'set_root_dir', 'reset_root_dir']
 
 _CONFIG_DIRNAME = 'catalog_configs'
 _GITHUB_URL = 'https://raw.githubusercontent.com/LSSTDESC/gcr-catalogs/master/GCRCatalogs'
+# Substitute for the following string if it starts a value for a path-like
+# keyword
+_ROOT_DIR_SIGNAL = "^/"
 _YAML_EXTENSIONS = ('.yaml', '.yml')
 
+# Keys appearing in yaml files whose values may be paths relative to _ROOT_DIR
+_PATH_LIKE_KEYS = ('filename', 'addon_filename', 'base_dir', 'root_dir',
+                   'catalog_root_dir', 'header_file', 'repo', 'table_dir')
+
+_DICT_LIST = ('catalogs')
```

Comment:
I think this should be a tuple, right?
```suggestion
_DICT_LIST = ('catalogs',)
```

Reply:
Right, but the issue is that, without the comma, `_DICT_LIST` would be a string rather than a tuple.
```python
>>> _DICT_LIST = ('catalogs')
>>> type(_DICT_LIST)
str
```